screen_tabs.py

```diff
@@ -285,10 +293,18 @@ def draw(self, context):
 					row.operator('scene.add_tab', text = '', icon='ZOOMIN')
 					row.alert = False
 				else:
+					if (addon_prefs.screen_text_length >= 0):
+						display_name	=	""
+						for i, a in enumerate(name):
```

Comment:
Better do this:
`if (addon_prefs.screen_text_length > 0): 
      name = name[0 : addon_prefs.screen_text_length]`

Reply:
I hardly know how colons :  work in lists, so I never considered it but that works. 

As for empty icons.....never thought about that.  
I'll set it to a checkbox

Reply:
Sorry, I may have misread/misunderstood  your point. 

There's a bug where if no name is displayed and no icon is available, where the boxes have weird sizes and the space is sort of blank.  
I thought this is what you meant.     
I opted   to use the 'BLANK1' icon that you use, to fix this


If you actually meant that it should just always display text, I disagree. 
When the value is set to -1, it defaults to displaying everything, and displaying the text at all is a personal preference which I can see someone preferring. 

It's like the hide labels thing for the task bar in Operating systems. 
Sometimes people want to have more stuff there, so hiding the text gives them a lot more space to do so. 
Others prefer to see each and every item available. 
I prefer to only see some stuff (so I modified registry to hide labels but don't combine windows)

Reply:
My point is that when you disable showing screen name and it doesn't have icon, what should it display?

Reply:
Ideally it would use a default icon from an enum property in the addons but that'd be a lot of text to setup. 

I thought about 'CHECKBOX_DEHLT' for inactive tabs and 'CHECKBOX_HLT' for the active tab,  or 'RADIOBUT_OFF' and 'RADIOBUT_ON',  but then I though what about 'FILE' or 'MESH_PLANE', or maybe 'FILE_HIDDEN'. 

deciding the best one to always use would always end up just a personal preference. 

So, I figured a blank button would just be universally better. 
Besides, anyone that would set the addon to hide names would probably choose to use custom icons for their scenes anyway  *(I do)*.